### ISSUE TITLE
"There was an Error: Error: socket hang up"

### ISSUE THREAD

**Issue by drmolotow** [dehsgr/node-red-contrib-meross#32]
Hi,
since the last Meross-Firmwareupdate for my MSS210 I get following error, when I try to turn on/off: 
`"There was an Error: Error: socket hang up"`

It's still possible to get the status. Is this a known issue?

**Comment by structure7** [comment]
Seeing the same here. It looks like updating smart plugs to Meross firmware 4.1.14 is breaking the node.

Digging a little more, perhaps they're encrypting or encoding their payloads now. 

Before:
```
{
  "payload": {
    "togglex": {
      "onoff": 0,
      "channel": 0
    }
  },
  "header": {
    "messageId": "5d3df30d8a4efc989a2dd2ef39b202e2",
    "method": "SET",
    "from": "http://192.168.0.211/config",
    "payloadVersion": 1,
    "namespace": "Appliance.Control.ToggleX",
    "uuid": "2109701483756151855648e1e96fd8f6",
    "sign": "37d52aed5e9ea978eee000a7c9aa771f",
    "triggerSrc": "iOSLocal",
    "timestamp": 1719202109
  }
}
```

After:
```
UCaTX4LHYHyqOyozSR/xG5xRAxXG6wTN4eghaCgOL1yU/aUNExB5jhNA8RwZTLU2InsEY0L4BIqXSJ0BLG6wYVGobnb1U81cYaZINcwY6BB0ghqb0fgIonm4dcXr9gzu3z40n8Iu2e/KyoSsMbwEkWNP9CDeBPi3tXEqtneWmKVrPaW8n6qp/gxThMIqbAe7IZ/zO3dajqNRIFMHio79lkVRz2gLc0rO4X83vSHTdgIK8xk31O4RV5gitBBux7j27WrI4z1ZDzQR/HdeGt2wduUPy+vYt6lVqwALiPUa2cLWVlKQ63sXUfrQ3cKw2uh2L8v5oZU99TpkVSlBj6mbO2Za5ZhKRsbcGbl6nGuCpQv2VPYC1NtOEVidZXCVkfabO8Rfqy54BNskdL7HNs9ZznppWlh5uvQBx0GFwBarF3i40onqs2MFKHpPqsKFcdIMVGvSSy3MSBET3G2V0PI91w==
```
Looks like it's back to Sonoff plugs!

**Comment by structure7** [comment]
Update: Looks like the Home Assistant gang is chewing on this: https://github.com/krahabb/meross_lan/issues/456

**Comment by dehsgr** [comment]
Thanks for your input. https://github.com/krahabb/meross_lan/commit/292001813cd597cd70d60991953d5b58e3f58d07 seems to be the solution for this. Currently there isn't much time for implementing a fix. I'll try my best to get it working again as soon as possible.

**Comment by dehsgr** [comment]
I'm on firmware 4.**`2`**.9. No issues here and no new firmware suggested from Meross for a test MSS210 (Hardware v4.0.0)?! @structure7 & @drmolotow what's your firmware version and hardware version and device type? @structure7 you told, firmware v4.**`1`**.14 is causing issue. But my version is higher?!

**Comment by dehsgr** [comment]
@structure7 & @drmolotow which version of this node you're using? Is it v2.3.1 (latest update) with key configured as mentioned [here](https://github.com/dehsgr/node-red-contrib-meross#howto-fetch-the-key-via-fetch-tool-requires-nodejs--v14)?

**Comment by structure7** [comment]
Ah sorry for the confusion. I'm using 2.3.1 but that firmware I mentioned (4.1.14) is what the Meross app said it was installing, which I think is incorrect! Regardless, the hardware is MSS110 with a mix of HW versions 4.0.0 and 7.0.0. The latest FW version I updated to was 7.3.37, which broke the node for me. The latest FW version I have that is still working is 7.3.35.

**Comment by dehsgr** [comment]
@structure7 I don‘t get firmware v7 offered. v4 is the latest firmware version for me. So I think I‘m not able to support here currently. :(

**Comment by structure7** [comment]
I had the 7.3.37 FW update to both HW 4 and 7 devices, and it broke them all for me.

**Comment by dehsgr** [comment]
Yeah, but I have to wait for v7 firmware for being able to support in this case. I‘m sorry. :(

**Comment by structure7** [comment]
Understood. I'm assuming the HW isn't a OTA update but rather what you get if you purchase a new device. Either way, thanks for the great node! It performed admirably for a very long time!

**Comment by dehsgr** [comment]
Hopefully they‘ll push that firmware out or someone contributes here. 

Another solution would be sending such a device to me with that firmware, or provide a system access where I can access node-red frontend and backend. 

All are ways for being able to get forward here. I‘ll do my best. In private I switched over to Shelly devices.

**Comment by structure7** [comment]
> Shelly devices

I forgot about those! I got a plug to try and I'm in love! Think I'll be switching to those too.

**Comment by Schischu** [comment]
Same error observed on mss315 with firmware 9.5.30 and mss110 with 7.3.46.
